src/ArrayView.hpp

```diff
@@ -303,7 +303,15 @@ class ArrayView
    */
   LVARRAY_HOST_DEVICE inline
   INDEX_TYPE size() const noexcept
-  { return indexing::multiplyAll< NDIM >( m_dims ); }
+  {
+    // Note: This used to be done with a recursive template but XL-release would produce incorrect results.
+    // Specifically in exampleArray it would return an "old" size even after being updated, strange.
+    INDEX_TYPE val = m_dims[ 0 ];
+    for( int i = 1; i < NDIM; ++i )
+    { val *= m_dims[ i ]; }
+
+    return val;
```

Comment:
```suggestion

#if defined( __ibmxl__ ) && defined( __CUDA_ARCH__ )
    INDEX_TYPE val = m_dims[ 0 ];
    for( int i = 1; i < NDIM; ++i )
    { val *= m_dims[ i ]; }

    return val;
#else
    return indexing::multiplyAll< NDIM >( m_dims );
#endif  
```

Reply:
I wanted to minimize the divergence between XL and the other compilers and my guess would be that both are equally fast, but I'd happily accept this change.

Reply:
I take the point of view that this is a problem with the XL compiler, so we should have to desired implementation, and the XL implementation.

Reply:
Done.